lib/board.rb

```diff
@@ -0,0 +1,92 @@
+class Board
+
+    def initialize()
+        @grid = Array.new(3){Array.new(3, '_')}
+    end
+
+    def get_grid
+        @grid
+    end
+
+    def find_winner()
+
+        # returns row winner
+        @grid.each do |row|
+            row_winner = find_winner_in_line(row)
+            if row_winner
+                return row_winner
+            end
+        end
+
+        # returns column winner
+        @grid.transpose.each do |col|
+            col_winner = find_winner_in_line(col)
+            if col_winner
+                return col_winner
+            end
+        end
+
+        # returns diagonal winner
+        diagonal_winner = find_winner_in_diagonal(@grid)
+        if diagonal_winner
+            return diagonal_winner
+        end
+
+        # check for any unfilled space
+        @grid.each do |row|
+            row.each do |col|
+                if col == '_'
```

Comment:
It might be helpful to make a private function to check if a cell is empty, so there's only one piece of code that has to remember that `"_"` is a blank cell.

```ruby
# where it is used
if is_empty?(col)
  return false
end

# where it is defined
private
def is_empty?(cell)
  cell == "_"
end
```